### PR TITLE
Test google result is redirected away from archive

### PR DESCRIPTION
**Pull request by BryanEliDimas**
Solves https://github.com/openstax/cnx/issues/315 **Create automated test for SEO google redirect** which indicates to test these code changes: https://github.com/openstax/cnx-deploy/pull/1084

### CODE REVIEW

File: tests/webview/test_google_redirect.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from requests import get
+from tests import markers
+
+
+@markers.webview
+@markers.nondestructive
+def test_google_redirect_archive_results(webview_base_url, chrome_options):
```
Comment (m1yag1) [inline tests/webview/test_google_redirect.py]: the items (fixtures) passed into this function are not being used. In your test function. I believe you should remove what is not needed. From what i can tell it's `chrome_options`

File: tests/webview/test_google_redirect.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    # end-to-end integration test for redirecting google away from archive URLs
+    # see: https://github.com/openstax/cnx/issues/209
+
+    url = "https://archive-qa.cnx.org/contents/fea3130c-6e57-41b2-a00f-0267ffae273c@5"
```
Comment (m1yag1) [inline tests/webview/test_google_redirect.py]: You should be able to build this with the proper fixture. `archive_base_url` should work.

Comment (m1yag1) [reply tests/webview/test_google_redirect.py]: You'll also want to parametrize the uuid and add an identifier.  A documentation reference is this: https://docs.pytest.org/en/latest/parametrize.html

Code examples within the framework are here:
https://github.com/openstax/cnx-automation/blob/master/tests/webview/ui/test_content.py#L218

The url can be provided by the `archive_base_url` fixture and example of that is here:
https://github.com/openstax/cnx-automation/blob/511e151d9e19adcb4288f05ce973e00b0372cd5c/tests/webview/ui/test_robots.py#L22

File: tests/webview/test_google_redirect.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    # TODO: do I need to clear/purge the varnish cache before running this test?
```
Comment (pumazi) [inline tests/webview/test_google_redirect.py]: This shouldn't be necessary. The varnish configuration doesn't cache the redirect response, right?

A configuration change to Varnish is going to restart/reload the service. So the cache would be flushed at that time. Changes to this behavior should be reflected by this test.

Comment (m1yag1) [reply tests/webview/test_google_redirect.py]: The TODO comment needs to be removed in that case.

Comment (BryanEliDimas) [reply tests/webview/test_google_redirect.py]: Done, removed comment.

File: tests/webview/test_google_redirect.py (hunk 2 above)
Comment (BryanEliDimas) [reply tests/webview/test_google_redirect.py]: Not exactly why, in this case, I need to parametrize but—done. 💯

File: tests/webview/test_google_redirect.py (hunk 1 above)
Comment (BryanEliDimas) [reply tests/webview/test_google_redirect.py]: Sorry, I missed it because I _was_ using them before.

**Comment by karenc** [comment]
@BryanEliDimas @pumazi @m1yag1 

**TL;DR** If the url is already cached in varnish, the google redirect does not work.

---

I had a look at varnishlog on staging and on qa to see what the problem was.  The python code I'm doing in my console looks like this:

```
Python 3.6.7 (default, Oct 22 2018, 11:32:17)
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import requests
>>> qa6 = requests.get('https://archive-qa.cnx.org/contents/a652edf4-562b-4629-a940-7fc35e8a3727@6', headers={'Referer': 'https://google.com/'})
>>> qa6.headers
{'Access-Control-Allow-Headers': 'origin,dnt,accept-encoding,accept-language,keep-alive,user-agent,x-requested-with,if-modified-since,cache-control,content-type', 'Access-Control-Allow-Methods': 'GET, OPTIONS', 'Access-Control-Allow-Origin': '*', 'Cache-Control': 'max-age=31536000, public', 'Content-Type': 'application/json', 'Date': 'Fri, 29 Mar 2019 21:38:30 GMT', 'Etag': '"b3eNPIrTgXJP7KLeVEBZ0w"', 'Expires': 'Sat, 28 Mar 2020 21:38:30 GMT', 'Link': '<https://qa.cnx.org/contents/caa57dab-41c7-455e-bd6f-f443cda5519c:a652edf4-562b-4629-a940-7fc35e8a3727/Introduction> ;rel="Canonical"', 'Server': 'waitress', 'status': '302', 'Location': 'https://qa.cnx.org/contents/caa57dab-41c7-455e-bd6f-f443cda5519c:a652edf4-562b-4629-a940-7fc35e8a3727/Introduction', 'X-Varnish-Status': 'uncacheable - redirected google', 'X-Varnish-Canonical-Url': 'https://qa.cnx.org/contents/caa57dab-41c7-455e-bd6f-f443cda5519c:a652edf4-562b-4629-a940-7fc35e8a3727/Introduction', 'X-Varnish-Backend': 'qa00_archive0', 'X-Varnish-Ttl': '60.000', 'X-Varnish': '623252', 'Age': '0', 'Via': '1.1 varnish-v4', 'X-Varnish-Cache': 'MISS', 'Accept-Ranges': 'bytes', 'Transfer-Encoding': 'chunked', 'Content-Encoding': 'gzip', 'Strict-Transport-Security': 'max-age=16000000'}
```

Meanwhile, I'm watching `sudo varnishlog -n varnish` on qa00.cnx.org:

```
*   << Request  >> 655829          
-   Begin          req 655828 rxreq                          
-   Timestamp      Start: 1553894597.760911 0.000000 0.000000
-   Timestamp      Req: 1553894597.760911 0.000000 0.000000
-   ReqStart       128.42.169.49 56792
-   ReqMethod      GET                                             
-   ReqURL         /contents/a652edf4-562b-4629-a940-7fc35e8a3727@6
-   ReqProtocol    HTTP/1.1                
-   ReqHeader      Host: archive-qa.cnx.org          
-   ReqHeader      User-Agent: python-requests/2.21.0
-   ReqHeader      Accept-Encoding: gzip, deflate
-   ReqHeader      Accept: */*                 
-   ReqHeader      Referer: https://google.com/
-   ReqHeader      X-Secure: true          
-   ReqHeader      X-Forwarded-Proto: https     
-   ReqHeader      X-Forwarded-For: 51.15.41.238
-   ReqHeader      Connection: close            
-   ReqUnset       X-Forwarded-For: 51.15.41.238               
-   ReqHeader      X-Forwarded-For: 51.15.41.238, 128.42.169.49
-   VCL_call       RECV          
-   VCL_acl        NO_MATCH block                
-   ReqUnset       Accept-Encoding: gzip, deflate
-   ReqHeader      Accept-Encoding: gzip
-   VCL_return     hash
-   VCL_call       HASH  
-   VCL_return     lookup
-   VCL_call       MISS 
-   VCL_return     fetch             
-   Link           bereq 655830 fetch                        
-   Timestamp      Fetch: 1553894597.843860 0.082948 0.082948
-   RespProtocol   HTTP/1.1
-   RespStatus     200
-   RespReason     OK
-   RespHeader     Access-Control-Allow-Headers: origin,dnt,accept-encoding,accept-language,keep-alive,user-agent,x-requested-with,if-modified-since,cache-control,content-type                                                            
-   RespHeader     Access-Control-Allow-Methods: GET, OPTIONS
-   RespHeader     Access-Control-Allow-Origin: *         
-   RespHeader     Cache-Control: max-age=31536000, public
-   RespHeader     Content-Length: 6707          
-   RespHeader     Content-Type: application/json     
-   RespHeader     Date: Fri, 29 Mar 2019 21:23:17 GMT
-   RespHeader     Etag: "b3eNPIrTgXJP7KLeVEBZ0w"        
-   RespHeader     Expires: Sat, 28 Mar 2020 21:23:17 GMT                                                                                                     
-   RespHeader     Link: <https://qa.cnx.org/contents/caa57dab-41c7-455e-bd6f-f443cda5519c:a652edf4-562b-4629-a940-7fc35e8a3727/Introduction> ;rel="Canonical"
-   RespHeader     Server: waitress
-   RespHeader     status: 302
-   RespHeader     Location: https://qa.cnx.org/contents/caa57dab-41c7-455e-bd6f-f443cda5519c:a652edf4-562b-4629-a940-7fc35e8a3727/Introduction
-   RespHeader     X-Varnish-Status: uncacheable - redirected google
-   RespHeader     X-Varnish-Canonical-Url: https://qa.cnx.org/contents/caa57dab-41c7-455e-bd6f-f443cda5519c:a652edf4-562b-4629-a940-7fc35e8a3727/Introduction
-   RespHeader     X-Varnish-Backend: qa00_archive0
-   RespHeader     X-Varnish-Ttl: 60.000
-   RespHeader     X-Varnish: 655829
-   RespHeader     Age: 0
-   RespHeader     Via: 1.1 varnish-v4
-   VCL_call       DELIVER
-   RespHeader     X-Varnish-Cache: MISS
-   VCL_return     deliver
-   Timestamp      Process: 1553894597.843958 0.083047 0.000099
-   RespHeader     Accept-Ranges: bytes
-   Debug          "RES_MODE 2"
-   RespHeader     Connection: close
-   Timestamp      Resp: 1553894597.844027 0.083116 0.000068
-   ReqAcct        294 0 294 1116 6707 7823
-   End
```

You see `X-Varnish-Status: uncacheable - redirected google` -- all good :+1:

Now request the latest version of the same page on staging:

```
>>> staging8 = requests.get('https://archive-staging.cnx.org/contents/a652edf4-562b-4629-a940-7fc35e8a3727@8', headers={'Referer': 'https://google.com/'})
>>> staging8.headers
{'Access-Control-Allow-Headers': 'origin,dnt,accept-encoding,accept-language,keep-alive,user-agent,x-requested-with,if-modified-since,cache-control,content-type', 'Access-Con
trol-Allow-Methods': 'GET, OPTIONS', 'Access-Control-Allow-Origin': '*', 'Cache-Control': 'max-age=31536000, public', 'Content-Type': 'application/json', 'Date': 'Fri, 01 Feb
 2019 20:08:52 GMT', 'Etag': '"oQaGRhncAcpek1I/pQN1xQ"', 'Expires': 'Sat, 01 Feb 2020 20:08:53 GMT', 'Link': '<https://staging.cnx.org/contents/caa57dab-41c7-455e-bd6f-f443cd
a5519c:a652edf4-562b-4629-a940-7fc35e8a3727/Introduction> ;rel="Canonical"', 'Server': 'waitress', 'X-Varnish-Status': 'cacheable - Cache-Control in backend response', 'X-Var
nish-Backend': 'staging07_archive4', 'X-Varnish-Ttl': '31536000.000', 'X-Varnish': '1848678 622783', 'Age': '4843622', 'Via': '1.1 varnish-v4', 'X-Varnish-Cache': 'HIT', 'Acc
ept-Ranges': 'bytes', 'Transfer-Encoding': 'chunked', 'Content-Encoding': 'gzip', 'Strict-Transport-Security': 'max-age=16000000'}
```

On staging01.cnx.org `sudo varnishlog -n varnish`:

```
*   << Request  >> 2204168
-   Begin          req 2204167 rxreq
-   Timestamp      Start: 1553895628.065000 0.000000 0.000000
-   Timestamp      Req: 1553895628.065000 0.000000 0.000000
-   ReqStart       128.42.169.22 59696
-   ReqMethod      GET
-   ReqURL         /contents/a652edf4-562b-4629-a940-7fc35e8a3727@8
-   ReqProtocol    HTTP/1.1
-   ReqHeader      Host: archive-staging.cnx.org
-   ReqHeader      User-Agent: python-requests/2.21.0
-   ReqHeader      Accept-Encoding: gzip, deflate
-   ReqHeader      Accept: */*
-   ReqHeader      Referer: https://google.com/
-   ReqHeader      X-Secure: true
-   ReqHeader      X-Forwarded-Proto: https
-   ReqHeader      X-Forwarded-For: 51.15.41.238
-   ReqHeader      Connection: close
-   ReqUnset       X-Forwarded-For: 51.15.41.238
-   ReqHeader      X-Forwarded-For: 51.15.41.238, 128.42.169.22
-   VCL_call       RECV
-   VCL_acl        NO_MATCH block
-   ReqUnset       Accept-Encoding: gzip, deflate
-   ReqHeader      Accept-Encoding: gzip
-   VCL_return     hash
-   VCL_call       HASH
-   VCL_return     lookup
-   Hit            622783
-   VCL_call       HIT
-   VCL_return     deliver
-   RespProtocol   HTTP/1.1
-   RespStatus     200
-   RespReason     OK
-   RespHeader     Access-Control-Allow-Headers: origin,dnt,accept-encoding,accept-language,keep-alive,user-agent,x-requested-with,if-modified-since,cache-control,content-type
-   RespHeader     Access-Control-Allow-Methods: GET, OPTIONS
-   RespHeader     Access-Control-Allow-Origin: *
-   RespHeader     Cache-Control: max-age=31536000, public
-   RespHeader     Content-Length: 7078
-   RespHeader     Content-Type: application/json
-   RespHeader     Date: Fri, 01 Feb 2019 20:08:52 GMT
-   RespHeader     Etag: "oQaGRhncAcpek1I/pQN1xQ"
-   RespHeader     Expires: Sat, 01 Feb 2020 20:08:53 GMT
-   RespHeader     Link: <https://staging.cnx.org/contents/caa57dab-41c7-455e-bd6f-f443cda5519c:a652edf4-562b-4629-a940-7fc35e8a3727/Introduction> ;rel="Canonical"
-   RespHeader     Server: waitress
-   RespHeader     X-Varnish-Status: cacheable - Cache-Control in backend response
-   RespHeader     X-Varnish-Backend: staging07_archive4
-   RespHeader     X-Varnish-Ttl: 31536000.000
-   RespHeader     X-Varnish: 2204168 622783
-   RespHeader     Age: 4843895
-   RespHeader     Via: 1.1 varnish-v4
-   VCL_call       DELIVER
-   RespHeader     X-Varnish-Cache: HIT
-   VCL_return     deliver
-   Timestamp      Process: 1553895628.065209 0.000209 0.000209
-   RespHeader     Accept-Ranges: bytes
-   Debug          "RES_MODE 2"
-   RespHeader     Connection: close
-   Timestamp      Resp: 1553895628.065340 0.000340 0.000131
-   ReqAcct        299 0 299 879 7078 7957
-   End
```

The request looks similar but the response is completely different: `X-Varnish-Status: cacheable - Cache-Control in backend response`.

It took me some time to figure out what's going on.  Staging had a cache hit, it's not going to the backend to fetch the content and just returned the cached content.  [The varnish change](https://github.com/openstax/cnx-deploy/commit/678549e5cade056d9c207eaa687124da56922ed6) looks like it might only take effect after the backend returns (it's in something called `vcl_backend_response`).  So if there's a cache hit, that path is not taken.

To check my theory, I did this:

```
>>> requests.get('https://archive-qa.cnx.org/contents/a652edf4-562b-4629-a940-7fc35e8a3727@6')
<Response [200]>
>>> requests.get('https://archive-qa.cnx.org/contents/a652edf4-562b-4629-a940-7fc35e8a3727@6', headers={'Referer': 'https://google.com/'})
<Response [200]>
>>> _.headers
{'Access-Control-Allow-Headers': 'origin,dnt,accept-encoding,accept-language,keep-alive,user-agent,x-requested-with,if-modified-since,cache-control,content-type', 'Access-Control-Allow-Methods': 'GET, OPTIONS', 'Access-Control-Allow-Origin': '*', 'Cache-Control': 'max-age=31536000, public', 'Content-Type': 'application/json', 'Date': 'Fri, 29 Mar 2019 22:01:04 GMT', 'Etag': '"b3eNPIrTgXJP7KLeVEBZ0w"', 'Expires': 'Sat, 28 Mar 2020 22:01:05 GMT', 'Link': '<https://qa.cnx.org/contents/caa57dab-41c7-455e-bd6f-f443cda5519c:a652edf4-562b-4629-a940-7fc35e8a3727/Introduction> ;rel="Canonical"', 'Server': 'waitress', 'X-Varnish-Status': 'cacheable - Cache-Control in backend response', 'X-Varnish-Backend': 'qa00_archive0', 'X-Varnish-Ttl': '31536000.000', 'X-Varnish': '338806 338804', 'Age': '3', 'Via': '1.1 varnish-v4', 'X-Varnish-Cache': 'HIT', 'Accept-Ranges': 'bytes', 'Transfer-Encoding': 'chunked', 'Content-Encoding': 'gzip', 'Strict-Transport-Security': 'max-age=16000000'}
```

.. and indeed the `X-Varnish-Status` is now `cacheable`.

**Comment by BryanEliDimas** [comment]
@karenc I am not sure how, but the response _is_ being cached but for only 60 seconds. Is this really a bug though, or a feature? 🤔

**Comment by BryanEliDimas** [comment]
@karenc I think I have the solution to the problem. Basically, move my code to the end.

```
[04/01 14:49:55] [master] /Users/bryaneli/openstax/cnx-deploy
iii > gd
diff --git a/roles/varnish/templates/etc/varnish/varnish.vcl b/roles/varnish/templates/etc/varnish/varnish.vcl
index 0da2734..4626ee1 100644
--- a/roles/varnish/templates/etc/varnish/varnish.vcl
+++ b/roles/varnish/templates/etc/varnish/varnish.vcl
@@ -404,16 +404,6 @@ sub vcl_backend_response {
                 set beresp.http.X-Varnish-Status = "uncacheable - legacy VersionedCompleteZip";
             }
 
-            # Redirect search traffic from archive.cnx.org URLs to the corresponding pages on CNX
-            # See: https://github.com/openstax/cnx/issues/209
-            if (bereq.http.host ~ "archive" && bereq.url ~ "^/contents/.*@.*" && bereq.http.Referer ~ "google") {
-              set beresp.http.status = 302;
-              set beresp.http.Location = regsub(beresp.http.Link, "^<(https://.*)>.*Canonical.", "\1");
-              set beresp.http.X-Varnish-Status = "uncacheable - redirected google";
-              # Expose the canonical url as diagnostic information
-              set beresp.http.X-Varnish-Canonical-Url = regsub(beresp.http.Link, "^<(https://.*)>.*Canonical.", "\1");
-            }
-
             # Serve stale requests for up to 1 minute longer than the ttl
             # Contact the backend to obtain a fresh response during this time
             set beresp.grace = 60s;
@@ -444,6 +434,17 @@ sub vcl_backend_response {
                 set beresp.ttl = 60s;
                 set beresp.http.Cache-Control = "public, max-age=60";
             }
+
+            # Redirect search traffic from archive.cnx.org URLs to the corresponding pages on CNX
+            # See: https://github.com/openstax/cnx/issues/209
+            if (bereq.http.host ~ "archive" && bereq.url ~ "^/contents/.*@.*" && bereq.http.Referer ~ "google") {
+              set beresp.http.status = 302;
+              set beresp.http.Location = regsub(beresp.http.Link, "^<(https://.*)>.*Canonical.", "\1");
+              set beresp.http.X-Varnish-Status = "uncacheable - redirected google";
+              set beresp.ttl = 0s;
+              # Expose the canonical url as diagnostic information
+              set beresp.http.X-Varnish-Canonical-Url = regsub(beresp.http.Link, "^<(https://.*)>.*Canonical.", "\1");
+            }
         }
     }
```

**Comment by BryanEliDimas** [comment]
@karenc Yes, I've tested it using the `dev` server and it seems to work just fine. The problem, I think is that this gets set *after* my code for the google redirect runs: 

```
set beresp.uncacheable = true;
set beresp.ttl = 60s;
```

and in the subsequent request, the following `if` statement evaluates to true, effectively skipping my code which inside the `else` block:

```
if (bereq.uncacheable) {
            set beresp.http.X-Varnish-Status = "uncacheable - pass or hit-for-pass";
        } else {
```

Or in your example, this statement evaluated to true and re-assigned the X-Varnish-Status variable:

```
elsif (beresp.http.Cache-Control) {
                set beresp.http.X-Varnish-Status = "cacheable - Cache-Control in backend response";
            }
```

While for the google redirect we don't care about either case, thus the Google redirect code should be at the end. Makes sense? I may not be explaining clearly.

**Comment by karenc** [comment]
@BryanEliDimas It's really great you found a fix already!  I wasn't sure what to do about it.  Maybe you can create a PR for cnx-deploy for the next release?

**Comment by BryanEliDimas** [comment]
@karenc Here is a PR against cnx-deploy for the fix: https://github.com/openstax/cnx-deploy/pull/1084

**Comment by BryanEliDimas** [comment]
@m1yag1 My test is passing, but Travis is failing because another—unrelated—test is failing. I've tried to reproduce it locally unsuccessfully. What should I do about this? The test that's failing is: `test_in_book_search`.

**Comment by m1yag1** [comment]
@BryanEliDimas DId you bring your branch up to date with master? I fixed the `test_in_book_search` test last week.

**Comment by karenc** [comment]
@m1yag1 Thanks!  I've rebased the branch and now it's working!  Is this ready to be merged?

**Comment by m1yag1** [comment]
@karenc It's definitely close! I have a few more questions. 

1. Does the # FIX ME comment need to be there? https://github.com/openstax/cnx-automation/pull/559/files#diff-d243cb365df94f4046c95f6c723fdc7eR15. If so, how do we plan to keep the changes in sync? In my experience, seldom do FIXME and TODO notes ever get addressed.
2. Is there a test case for this in TestRail? The reason I ask is we run automated tests and upload their results to TestRail so we can see the results in our dashboard. It also helps us collect metrics on test failures. cc: @omehes 
3. Would we want to add an integration test marker to isolate this test? Pytest supports binary operators in the `-m` argument. We could run only these tests with `-m "webview and integration"` or `-m integration`

**Comment by BryanEliDimas** [comment]
@m1yag1 
1. Not anymore, I've removed it in my last (force-pushed) commit.
2. I don't know anything about TestRails, I don't know how to access it, but I could give it a shot :). 
3. That would be your call :), but I personally don't really see that much benefit to being able to run only integration tests.
4. I don't fully understand the meaning of "smoke test run" in the context of our QA process, but outside OpenStax QA context, I wouldn't say that it's a smoke test. It's a pretty complete test which requires integration between archive, Varnish, and webview.

Note that in my most recent commit, I also rename the file and the test, rephrased a comment, and uncommented a line of code. Oh, also btw, there has been a deployment to the QA server after Karen and I confirmed that this test works in that server.

**Comment by m1yag1** [comment]
@BryanEliDimas 
The smoke tests are a small set of the entire test suite we run to quickly check that everything is operating normally. They're nice b/c you can be pretty confident things are operating as they should. Due to this testing redirects it seemed like a good candidate. Smoke testing comes from "old school" hardware testing where you would plug in some electronic component and if it didn't cause smoke that was a good thing 😂 .

File: tests/webview/integration/test_webview_integration.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+@markers.webview
+@markers.nondestructive
+@markers.parametrize("uuid, version", [("fea3130c-6e57-41b2-a00f-0267ffae273c", 5)])
+def test_archive_redirect_for_google_results(archive_base_url, uuid, version):
+    # end-to-end integration test for redirecting google away from archive URLs
+    # see: https://github.com/openstax/cnx/issues/209
+
+    # GIVEN an "archive" URL
+    url = f"https://archive-qa.cnx.org/contents/{uuid}@{version}"
```
Comment (m1yag1) [inline tests/webview/integration/test_webview_integration.py]: Since this is now in staging can we go ahead and use the `archive_base_url` fixture?

Comment (m1yag1) [reply tests/webview/integration/test_webview_integration.py]: `https://{archive_base_url}/contents/{uuid}@{version}`

Comment (BryanEliDimas) [reply tests/webview/integration/test_webview_integration.py]: It is in staging?? It's _just_ been merged: https://github.com/openstax/cnx-deploy/pull/1084

Comment (m1yag1) [reply tests/webview/integration/test_webview_integration.py]: plz update the f-string to use the `archive_base_url`.

Comment (BryanEliDimas) [reply tests/webview/integration/test_webview_integration.py]: Done.

**Comment by m1yag1** [comment]
@BryanEliDimas 
Yikes, this failed in staging. Wasn't this deployed to staging? Or, did I catch that wrong?

**Comment by BryanEliDimas** [comment]
@m1yag1 It's not in staging yet. I did mention it 😅 https://github.com/openstax/cnx-automation/pull/559#discussion_r274607733

**Comment by m1yag1** [comment]
> @m1yag1 It's not in staging yet. I did mention it 😅 [#559 (comment)](https://github.com/openstax/cnx-automation/pull/559#discussion_r274607733)

@BryanEliDimas 
Ah sorry about that... you did say that. i had thought you said staging. We can merge this once the fix is in staging. I'll add the blocked label until then.

**Comment by m1yag1** [comment]
@pumazi plz add your review too. I'm not an expert on neb. ;)

**Comment by BryanEliDimas** [comment]
@m1yag1 The code that this PR/test is testing should've been deploy since Monday. But something went wrong (someone made a mistake when tagging `cnx-deploy`), therefore it has not been deployed yet and I don't know when it will be.

**Comment by BryanEliDimas** [comment]
@m1yag1 The code has been deployed to staging and the tests on Travis now pass :). This PR is ready to be merged! 🚀
